src/electron/device-connect-view/components/device-connect-view-container.tsx

```diff
@@ -36,18 +42,30 @@ export class DeviceConnectViewContainer extends React.Component<DeviceConnectVie
                 <WindowTitle title={brand}>
                     <BrandBlue />
                 </WindowTitle>
-                <DeviceConnectBody deps={this.props.deps} />
+                <DeviceConnectBody
+                    deps={this.props.deps}
+                    viewState={{
+                        deviceConnectState: this.state.deviceStoreData.deviceConnectState,
+                        connectedDevice: this.state.deviceStoreData.connectedDevice,
+                    }}
+                />
                 <TelemetryPermissionDialog deps={this.props.deps} isFirstTime={this.state.userConfigurationStoreData.isFirstTime} />
             </>
         );
     }
 
     public componentDidMount(): void {
-        this.props.deps.userConfigurationStore.addChangedListener(() => {
-            const newState = this.props.deps.userConfigurationStore.getState();
-            this.setState({
-                userConfigurationStoreData: newState,
-            });
+        this.addChangeListener('deviceStore');
+        this.addChangeListener('userConfigurationStore');
+    }
+
+    private addChangeListener(storeName: keyof DeviceConnectViewContainerStoreDeps): void {
```

Comment:
can we leverage BaseClientStoresHub? this code seems to be duplicating some of the functionality of BaseClientStoresHub

Reply:
Right! We should be able to do that. I'll udpate this PR with those changes.

Reply:
cool. other than that, LGTM

Reply:
Updated